docs/content/hasher.md

```diff
@@ -0,0 +1,188 @@
+---
+title: "Hasher"
+description: "Handle checksums for other remotes"
+---
+
+# {{< icon "fa fa-check-double" >}}Hasher (EXPERIMENTAL)
+
+The `hasher` overlay caches hashsums for underlying remote.
+
+To use it, first set up the underlying remote following the configuration
+instructions for that remote. You can also use a local pathname instead of
+a remote.
+
+First check your chosen remote is working - we'll call it `remote:path` here.
+Note that anything inside `remote:path` will be handled and anything outside
+won't. This means that if you are using a bucket based remote (e.g. S3, B2, swift)
+then you should probably put the bucket in the remote `s3:bucket`.
+
+Now configure `hasher` remote using `rclone config`.
+
+```
+No remotes found - make a new one
+n) New remote
+s) Set configuration password
+q) Quit config
+n/s/q> n
+name> hasher
+Type of storage to configure.
+Choose a number from below, or type in your own value
+[snip]
+XX / Cache hashsums
+   \ "hasher"
+[snip]
+Storage> hasher
+Remote to cache hashsums for, like "remote:path".
+Enter a string value. Press Enter for the default ("").
+remote> remote:path
+Choose supported hashsums.
+Enter a string value. Press Enter for the default ("md5").
+hashsums> md5
+Edit advanced config? (y/n)
+y) Yes
+n) No
+y/n> n
+Remote config
+--------------------
+[hasher]
+type = hasher
+remote = remote:path
+hashsums = md5
+--------------------
+y) Yes this is OK
+e) Edit this remote
+d) Delete this remote
+y/e/d> y
+```
+
+### Specifying the remote
+
+Make sure the remote has a `:` in normally. If you specify the remote
+without a `:` then rclone will use a local directory of that name.
+So if you use a remote of `/path/to/secret/files` then rclone will
+handle hashes for that directory. If you use a remote of `name` then rclone
+will put files in a directory called `name` in the current directory.
+
+### Hashsums
+
+If the base remote can instantly produce a checksum, hasher will
+transparently redirect hash requests to it.
+
+Many storage backends support MD5 and SHA1 hash types, so does chunker.
+With chunker you can choose one or another but not both.
+MD5 is set by default as the most supported type.
+Since chunker keeps hashes for composite files and falls back to the
+wrapped remote hash for non-chunked ones, we advise you to choose the same
+hash type as supported by wrapped remote so that your file listings
+look coherent.
+
+If your storage backend does not support MD5 or SHA1 but you need consistent
```

Comment:
Is this bit of the doc out of date?

Reply:
i haven't yet worked on docs.

Reply:
I noted above that docs are not yet ready. What you see is a stub (I guess I copy-pasted it from chunker). I'll let you know when I do the docs.